pkg/ruler/ruler.go

```diff
@@ -476,6 +491,29 @@ func instanceOwnsRuleGroup(r ring.ReadRing, g *rulespb.RuleGroupDesc, disabledRu
 	return ownsRuleGroup, nil
 }
 
+func instanceBacksUpRuleGroup(r ring.ReadRing, g *rulespb.RuleGroupDesc, disabledRuleGroups validation.DisabledRuleGroups, instanceAddr string) (bool, error) {
```

Comment:
This function seems does the same as `instanceOwnsRuleGroup` except using a different condition.
One is `ownsRuleGroup := rlrs.Instances[0].Addr == instanceAddr` and this one is 

```
	var backupRuleGroup bool
	// Only the second up to the last replica is used a backup
	for i := 1; i < len(rlrs.Instances); i++ {
		if rlrs.Instances[i].Addr == instanceAddr {
			backupRuleGroup = true
			break
		}
	}
```

Can we just make this condition logic a parameter and remove duplicate code?